bleach/html5lib_shim.py

```diff
@@ -232,7 +233,21 @@ def __iter__(self):
 
         for token in super(BleachHTMLTokenizer, self).__iter__():
             if last_error_token is not None:
-                if ((last_error_token['data'] == 'expected-closing-tag-but-got-char' and
+                if ((last_error_token['data'] == 'invalid-character-in-attribute-name' and
+                     token['type'] in TAG_TOKEN_TYPES and
+                     token.get('data'))):
+                    # Remove attribute names that have ', " or < in them
+                    # because those characters are invalid for attribute names.
+                    token['data'] = [
+                        item for item in token['data']
+                        if ('"' not in item[0] and
+                            "'" not in item[0] and
+                            '<' not in item[0])
+                    ]
+                    last_error_token = None
+                    yield token
```

Comment:
What this PR does is detect when the `HTMLTokenizer` raises its eyebrows about a bad attribute name and then remove attributes with invalid names.

Relevant spec: https://html.spec.whatwg.org/multipage/syntax.html#syntax-attribute-name